back/node_watcher/src/nodeWatcher.ts

```diff
@@ -81,11 +81,19 @@ async function incrementor(
       api.registry.setMetadata(rpcMeta);
     }
 
+    const events = await api.query.system.events.at(blockHash);
+    const sessionIndex = await api.query.session.currentIndex.at(blockHash);
```

Comment:
optimization inside optimization: could use Promise.all!

Reply:
wait, even api.queryMulti?

Reply:
hmm I wonder which one of those would be better actually

Reply:
queryMulti: it sends one WS message. Promise.all send 2.

Reply:
ah cool 👌

Reply:
how do you queryMulti on `.at` decorated functions? this is erroring for me:
```
    api.queryMulti([
      [api.query.system.events.at, blockHash],
      [api.query.session.currentIndex.at, blockHash]
    ], ([events, sessionIndex]) => {
      
    })
```

but this works
```
    api.queryMulti([
      api.query.system.events,
      api.query.session.currentIndex
    ], ([events, index]) => {})
```

Reply:
FWICT `StorageEntryPromiseOverloads` does not expect the decorations

Reply:
ah, my bad, you cannot do queryMulti and .at, see https://github.com/polkadot-js/api/issues/1039

So Promise.all it is